src/main/java/io/antmedia/rtmp/InProcessRtmpPublisher.java

```diff
@@ -0,0 +1,212 @@
+                IoBuffer flvPayload = IoBuffer.allocate(5 + nioBuf.remaining());
+                flvPayload.setAutoExpand(false);
+
+                // FrameType (key/inter) + CodecID (7 = AVC)
+                int frameType = (packet.flags() & avcodec.AV_PKT_FLAG_KEY) != 0 ? 0x10 : 0x20; // KEY=1(interleaved) but shift later
+                frameType = ((packet.flags() & avcodec.AV_PKT_FLAG_KEY) != 0) ? 0x17 : 0x27; // 1:keyframe|7=AVC , 2:inter frame|7
+                flvPayload.put((byte) frameType);
+                flvPayload.put((byte) 0x01);             // AVC NALU packet
+                flvPayload.put((byte) 0x00);             // composition time 0
+                flvPayload.put((byte) 0x00);
+                flvPayload.put((byte) 0x00);
+                flvPayload.put(nioBuf);
+                flvPayload.flip();
+
+                VideoData videoData = new VideoData(flvPayload);
+                videoData.setTimestamp(ts);
+                RTMPMessage rtmp = RTMPMessage.build(videoData, Constants.SOURCE_TYPE_LIVE);
+                broadcastScope.pushMessage(rtmp);
+            } else if (codecType == AVMEDIA_TYPE_AUDIO) {
+                int ts = (int) avutil.av_rescale_q(packet.pts(), audioTb, MuxAdaptor.TIME_BASE_FOR_MS);
+
+                ByteBuffer nioBuf = packet.data().limit(packet.size()).asByteBuffer();
+                IoBuffer flvPayload = IoBuffer.allocate(2 + nioBuf.remaining());
+                flvPayload.setAutoExpand(false);
+                // SoundFormat (10 = AAC)  | SoundRate 3=44k  | SoundSize 1=16-bit | SoundType 1=stereo
+                flvPayload.put((byte) 0xAF);
+                flvPayload.put((byte) 0x01); // AAC raw
+                flvPayload.put(nioBuf);
+                flvPayload.flip();
+
+                AudioData audioData = new AudioData(flvPayload);
+                audioData.setTimestamp(ts);
+                RTMPMessage rtmp = RTMPMessage.build(audioData, Constants.SOURCE_TYPE_LIVE);
+                broadcastScope.pushMessage(rtmp);
+            }
+        } catch (Exception e) {
+            logger.error("Error while pushing video packet", e);
+        }
+    }
+
+    public IBroadcastScope attachRtmpPublisher(String streamId) {
+        // obtain or create broadcast scope
+        IBroadcastScope bs = this.scope.getBroadcastScope(streamId);
+        if (bs == null) {
+            bs = new BroadcastScope(this.scope, streamId);
+            this.scope.addChildScope(bs);
+        }
+
+        // connect provider to scope pipe
+        bs.subscribe(this, null);
+
+        logger.info("In-process RTMP pipeline ready for stream {}", streamId);
+        return bs;
+    }
+
+    public void detachRtmpPublisher(String streamId) {
+        IBroadcastScope bs = this.scope.getBroadcastScope(streamId);
+        if (bs != null) {
+            bs.unsubscribe(this);
+        }
+    }
+
+    @Override
+    public boolean isCodecSupported(int codecId) {
+        return true;
```

Comment:
I suggest to implement this method to support H264 and AAC only. It may cause adding multiple codecs or wrong codecs when there is a multibitrate streaming